exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/__init__.py

```diff
@@ -0,0 +1,354 @@
+    ) -> TimeSeries:
+        name = last_value_record.instrument.name
+        value = last_value_record.aggregator.checkpoint
+        return [self.create_timeseries(last_value_record, name, value)]
+
+    def convert_from_value_observer(
+        self, value_observer_record: ExportRecord
+    ) -> TimeSeries:
+        timeseries = []
+        agg_types = ["min", "max", "sum", "count", "last"]
+        for agg_type in agg_types:
+            name = value_observer_record.instrument.name + "_" + agg_type
+            value = getattr(
+                value_observer_record.aggregator.checkpoint, agg_type
+            )
+            timeseries.append(
+                self.create_timeseries(value_observer_record, name, value)
+            )
+        return timeseries
+
+    # TODO: Implement convert from quantile once supported by SDK for Prometheus Summaries
+    def convert_from_quantile(
+        self, summary_record: ExportRecord
+    ) -> TimeSeries:
+        raise NotImplementedError()
+
+    # pylint: disable=no-member
+    def create_timeseries(
+        self, export_record: ExportRecord, name, value: float
+    ) -> TimeSeries:
+        timeseries = TimeSeries()
+        # Add name label, record labels and resource labels
+        timeseries.labels.append(self.create_label("__name__", name))
+        resource_attributes = export_record.resource.attributes
+        for label_name, label_value in resource_attributes.items():
+            timeseries.labels.append(
+                self.create_label(label_name, label_value)
+            )
+        for label in export_record.labels:
+            if label[0] not in resource_attributes.keys():
+                timeseries.labels.append(self.create_label(label[0], label[1]))
+        # Add sample
+        timeseries.samples.append(
+            self.create_sample(
+                export_record.aggregator.last_update_timestamp, value
+            )
+        )
+        return timeseries
+
+    def create_sample(self, timestamp: int, value: float) -> Sample:
+        sample = Sample()
+        sample.timestamp = int(timestamp / 1000000)
+        sample.value = value
+        return sample
+
+    def create_label(self, name: str, value: str) -> Label:
+        label = Label()
+        # Label name must contain only alphanumeric characters and underscores
+        label.name = re.sub("[^0-9a-zA-Z_]+", "_", name)
+        label.value = value
+        return label
+
+    def build_message(self, timeseries: Sequence[TimeSeries]) -> bytes:
+        write_request = WriteRequest()
+        write_request.timeseries.extend(timeseries)
+        serialized_message = write_request.SerializeToString()
+        return snappy.compress(serialized_message)
+
+    def get_headers(self) -> Dict:
+        headers = {
+            "Content-Encoding": "snappy",
+            "Content-Type": "application/x-protobuf",
+            "X-Prometheus-Remote-Write-Version": "0.1.0",
+        }
+        if self.headers:
```

Comment:
This is pretty confusing, the method is named `get_headers` but it also sets `self.headers`. Maybe this method should be split?

Reply:
Renamed the method to build_headers since that's what it is really doing. The old name did seem like a getter.